techtv2ovs/utils.py

```diff
@@ -0,0 +1,314 @@
+        self.db_pw = db_pw
+        self.db_name = db_name
+        self.db_host = db_host
+        self.collection = collection
+        self.aws = aws
+        self.protected = protected
+        self.output = output
+
+    def run(self):
+        """
+        Run the import
+        """
+        with MySQLdb.connect(
+            user=self.db_user, password=self.db_pw, db=self.db_name, host=self.db_host
+        ) as self.connection:
+            bonobo.settings.LOGGING_LEVEL.set(logging.ERROR)
+            bonobo.run(
+                self.etl_graph()
+            )
+
+    def process_thumbs(self, ttv_video):
+        """
+        Copy over all 'jumbo' thumbnails from TechTV to OVS buckets for a TechTVVideo.
+        Create a VideoThumbnail object for each.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo to process
+        """
+        ttv_video.thumbnail_status = ImportStatus.CREATED
+        s3client = boto3.client('s3')
+        s3list = s3client.list_objects_v2(Bucket=TTV_THUMB_BUCKET, Prefix=ttv_video.external_id)
+        if 'Contents' not in s3list or not s3list['Contents']:
+            ttv_video.thumbnail_status = ImportStatus.MISSING
+            ttv_video.save()
+            return
+        user_list = [thumb for thumb in s3list['Contents'] if '{}/jumbo'.format(ttv_video.ttv_id) in thumb['Key']]
+        default_list = [thumb for thumb in s3list['Contents'] if 'thumbnails/jumbo' in thumb['Key']]
+        idx = 0
+        for thumb in user_list + default_list:
+            new_key = thumb['Key'].split('/')[-1].replace('jumbo', str(idx))
+            try:
+                src = {'Bucket': TTV_THUMB_BUCKET, 'Key': thumb['Key']}
+                dst_bucket = settings.VIDEO_S3_THUMBNAIL_BUCKET
+                dst_key = 'thumbnails/techtv/{}/{}'.format(ttv_video.video.hexkey, new_key)
+
+                if self.aws:
+                    s3client.copy(src, dst_bucket, dst_key)
+
+                VideoThumbnail.objects.get_or_create(s3_object_key=dst_key, defaults={
+                    'bucket_name': dst_bucket,
+                    'video': ttv_video.video
+                })
+                idx += 1
+            except Exception as exc:
+                self.output.write("Error: thumbnail {} for ttv video {}".format(src['Key'], ttv_video.ttv_id))
+                ttv_video.status = ImportStatus.ERROR
+                ttv_video.thumbnail_status = ImportStatus.ERROR
+                ttv_video.errors += '{}\n\n'.format(str(exc))
+
+        if ttv_video.thumbnail_status != ImportStatus.ERROR:
+            ttv_video.thumbnail_status = ImportStatus.COMPLETE
+        ttv_video.save()
+
+    def process_files(self, ttv_video, s3videos):
+        """
+        Import videofiles, thumbnails, and captions for a TechTVVideo.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo object to process
+            s3videos (list): List of video file S3 objects
+        """
+        self.process_thumbs(ttv_video)
```

Comment:
Can you write a test covering this function?

Reply:
✅